src/awkward/_v2/contents/recordarray.py

```diff
@@ -824,30 +824,29 @@ def _reduce_next(
         outlength,
         mask,
         keepdims,
+        behavior,
     ):
-        contents = []
-        for content in self._contents:
-            contents.append(
-                content[: self._length]._reduce_next(
-                    reducer,
-                    negaxis,
-                    starts,
-                    shifts,
-                    parents,
-                    outlength,
-                    mask,
-                    keepdims,
+        reducer_name = reducer.__name__.split(".")[0].lower()
+
+        if behavior is None or not (
+            any(
+                x[0].__module__.split(".")[-1].split("_")[-1] == reducer_name
+                for x in behavior.keys()
+            )
+        ):
```

Comment:
This test should be implemented in _util.py, similar to

https://github.com/scikit-hep/awkward/blob/9e8c841cd9722efb5211040b0c476710706de69d/src/awkward/_v2/_util.py#L590-L597

and the other functions surrounding it. The `reducer` object isn't guaranteed to have the same name as the `ak.*` function (though right now that happens to be true); we should do it by referential identity matching the function itself. I guess we could attach each of those functions to the reducer in question.

For instance, at the end of each file like https://github.com/scikit-hep/awkward/blob/main/src/awkward/_v2/operations/ak_argmin.py, add

```python
ak._v2._reducers.ArgMin.function = argmin
```

(and similarly for argmin, argmax, count, count_nonzero, sum, prod, any, all, min, max). Then, we can always get the high-level function from the `reducer`, and that's what we need to look up in `behavior`. The second part is `self.parameter("__record__")`, which is not allowed to be `None`.

As a helper function in [_util](https://github.com/scikit-hep/awkward/blob/main/src/awkward/_v2/_util.py), this could look like

```python
def reducer_recordclass(reducer, layout, behavior):
    behavior = Behavior(ak._v2.behavior, behavior)
    rec = layout.parameter("__record__")
    if isstr(rec):
        return behavior[reducer.function, rec]
```

If `ak._v2._util.reducer_recordclass(reducer, self, behavior)` returns `None`, then you should raise the `TypeError`. If it's not `None`, then you have the `NotImplementedError` for now, as a placeholder for future implementation.

Note that this `Behavior` is a way of merging local array `behavior` with the global `ak.behavior` that should be applied consistently for all behavior-lookups, and its `__getitem__` would return `None` if there's no match (so both code paths in `reducer_recordclass` can return `None`).